File: 3/3a.py
with open("3.txt") as file:
    backpacks = file.read().split('\n')
    res = 0
    for backpack in backpacks:
        counter = dict()
        for i in range(int(len(backpack)/2)):
            counter[backpack[i]] = 1
        for i in range(int(len(backpack)/2), len(backpack)):
            if backpack[i] in counter:
                ch = backpack[i]
                chVal = ord(ch)
                if chVal >= 97:
                    res += chVal - ord('a') + 1
                    break
                else:
                    res += chVal - ord('A') + 27
                    break
    print(res)

File: 3/3a.js
const fs = require('fs');

const input = fs.readFileSync('./3.txt', 'utf8');

const backpacks = input.split(/\r?\n/);

let res = 0;

for (const backpack of backpacks) {
  const counter = {};
  for (let i = 0; i < backpack.length / 2; i++) {
    counter[backpack[i]] = 1;
  }
  for (let i = backpack.length / 2; i < backpack.length; i++) {
    if (counter[backpack[i]] === 1) {
      const ch = backpack[i];
      const chVal = ch.charCodeAt();
      if (chVal >= 97) {
        res += chVal - 'a'.charCodeAt() + 1;
        break;
      } else {
        res += chVal - 'A'.charCodeAt() + 27;
        break;
      }
    }
  }
}

console.log(res);
